src/executor/stack/memory.rs

```diff
@@ -393,6 +415,17 @@ impl<'config> MemoryStackSubstate<'config> {
 	pub fn touch<B: Backend>(&mut self, address: H160, backend: &B) {
 		self.account_mut(address, backend);
 	}
+
+	pub fn get_tstorage(&mut self, address: H160, key: H256) -> Result<U256, ExitError> {
+		self.tstorages
+			.get(&(address, key))
+			.cloned()
+			.ok_or(ExitError::InvalidRange)
```

Comment:
@birchmd thanks for the the useful documentation link.
I refactored as you suggested.

Reply:
Thanks for the changes. Please also add code to the `exit_commit` implementation in `MemoryStackSubstate` to append the `tstorages` into the parent. Otherwise the `tstorages` will get lost when a call to a contract completes.

Reply:
Added in da568d3